libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -165,6 +167,26 @@ void AC_PosControl::set_alt_target_from_climb_rate(float climb_rate_cms, float d
         _pos_target.z += climb_rate_cms * dt;
     }
 
+#if APM_BUILD_TYPE(APM_BUILD_ArduSub)
```

Comment:
I think an AC_PosControl_Sub class that inherits from AC_PosControl would be better. I'm sure you will want to add more features later, and that would provide an cleaner separation